internal/operands/tekton-tasks/tekton-tasks.go

```diff
@@ -0,0 +1,313 @@
+	return []client.Object{
+		&rbac.ClusterRole{},
+		&pipeline.ClusterTask{},
+		&rbac.RoleBinding{},
+		&v1.ServiceAccount{},
+	}
+}
+
+func (t *tektonTasks) WatchTypes() []client.Object {
+	return nil
+}
+
+func (t *tektonTasks) RequiredCrds() []string {
+	return requiredCRDs
+}
+
+func (t *tektonTasks) filterUnusedObjects() {
+	newTasks := []pipeline.Task{}
+	for _, task := range t.tasks {
+		if _, ok := AllowedTasks[task.Name]; ok {
+			newTasks = append(newTasks, task)
+		}
+	}
+	t.tasks = newTasks
+
+	newSA := []v1.ServiceAccount{}
+	for _, sa := range t.serviceAccounts {
+		if _, ok := AllowedTasks[strings.TrimSuffix(sa.Name, "-task")]; ok {
+			newSA = append(newSA, sa)
+		}
+	}
+	t.serviceAccounts = newSA
+
+	newRB := []rbac.RoleBinding{}
+	for _, rb := range t.roleBindings {
+		if _, ok := AllowedTasks[strings.TrimSuffix(rb.Name, "-task")]; ok {
+			newRB = append(newRB, rb)
+		}
+	}
+	t.roleBindings = newRB
+
+	newCR := []rbac.ClusterRole{}
+	for _, cr := range t.clusterRoles {
+		if _, ok := AllowedTasks[strings.TrimSuffix(cr.Name, "-task")]; ok {
+			newCR = append(newCR, cr)
+		}
+	}
+	t.clusterRoles = newCR
+}
+
+func (t *tektonTasks) Reconcile(request *common.Request) ([]common.ReconcileResult, error) {
+	var results []common.ReconcileResult
+	var reconcileFunc []common.ReconcileFunc
+	reconcileFunc = append(reconcileFunc, reconcileTektonTasksFuncs(t.tasks)...)
+	reconcileFunc = append(reconcileFunc, reconcileClusterRoleFuncs(t.clusterRoles)...)
+	reconcileFunc = append(reconcileFunc, reconcileServiceAccountsFuncs(t.serviceAccounts)...)
+	reconcileFunc = append(reconcileFunc, reconcileRoleBindingFuncs(t.roleBindings)...)
+
+	reconcileTektonBundleResults, err := common.CollectResourceStatus(request, reconcileFunc...)
+	if err != nil {
+		return nil, err
+	}
+
+	upgradingNow := isUpgradingNow(request)
+	for _, r := range reconcileTektonBundleResults {
+		if !upgradingNow && (r.OperationResult == common.OperationResultUpdated) {
+			request.Logger.Info(fmt.Sprintf("Changes reverted in tekton tasks: %s", r.Resource.GetName()))
+		}
+	}
+	return append(results, reconcileTektonBundleResults...), nil
+}
+
+func (t *tektonTasks) Cleanup(request *common.Request) ([]common.CleanupResult, error) {
+	var objects []client.Object
+
+	for _, t := range t.tasks {
+		o := t.DeepCopy()
+		objects = append(objects, o)
+	}
+	for _, cr := range t.clusterRoles {
+		o := cr.DeepCopy()
+		objects = append(objects, o)
+	}
+	for _, rb := range t.roleBindings {
+		o := rb.DeepCopy()
+		objects = append(objects, o)
+	}
+	for _, sa := range t.serviceAccounts {
+		o := sa.DeepCopy()
+		objects = append(objects, o)
+	}
+
+	clusterTasks, err := listDeprecatedClusterTasks(request)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, ct := range clusterTasks {
+		o := ct.DeepCopy()
+		objects = append(objects, o)
+	}
+
+	return common.DeleteAll(request, objects...)
+}
+
+func listDeprecatedClusterTasks(request *common.Request) ([]pipeline.ClusterTask, error) {
```

Comment:
@ksimon1, This function gets all ClusterTasks with a specific label. Why are they considered deprecated?

Reply:
The clusterTask object is deprecated https://tekton.dev/docs/pipelines/tasks/#task-vs-clustertask

Reply:
Ok. @codingben, can you please add a comment in code explaining that `ClusterTask` resource is deprecated?

Reply:
Yes, inside it, or as a docstring above the function.

Reply:
Done.